index.js

```diff
@@ -55,15 +55,17 @@ function rateLimitPlugin (fastify, settings, next) {
   // onRoute add the preHandler rate-limit function if needed
   fastify.addHook('onRoute', (routeOptions) => {
     if (routeOptions.config) {
-      if (routeOptions.config.rateLimit && typeof routeOptions.config.rateLimit === 'object') {
-        const current = Object.create(pluginComponent)
-        current.store = pluginComponent.store.child(routeOptions)
-        // if the current endpoint have a custom rateLimit configuration ...
-        buildRouteRate(current, makeParams(routeOptions.config.rateLimit), routeOptions)
-      } else if (routeOptions.config.rateLimit === false) {
-        // don't apply any rate-limit
-      } else {
-        throw new Error('Unknown value for route rate-limit configuration')
+      if (routeOptions.config.rateLimit) {
+        if (typeof routeOptions.config.rateLimit === 'object') {
+          const current = Object.create(pluginComponent)
+          current.store = pluginComponent.store.child(routeOptions)
+          // if the current endpoint have a custom rateLimit configuration ...
+          buildRouteRate(current, makeParams(routeOptions.config.rateLimit), routeOptions)
+        } else if (routeOptions.config.rateLimit === false) {
+          // don't apply any rate-limit
```

Comment:
I don't think this condition could actually be triggered, because `routeOptions.config.rateLimit` is already going to be truthy. Would you mind removing it?

Reply:
Hi @mcollina you're right, in fact, my fix solves some issues, but not all. The problem is that all tests are made without the assumption of existing per-route configuration for other plugins.

For example, if I add a route config to the other tests, most of them will fail even with my fix.

I will submit an updated PR later today that solves all use-cases...